### ISSUE TITLE
Java version errors when users use Java > 17

### ISSUE THREAD

**Issue by alexarchambault** [VirtusLab/scala-cli#1083]
When running Scala CLI on a new machine and letting it use Java 17 for Bloop, if users later on manually download and use Java 18, it seems Bloop or javac / scalac complain about a too high Java version being requested when compiling things.

See discussion on Discord around @Baccata's message [here](https://discord.com/channels/632150470000902164/901153021638082660/983693710036459530).

**Comment by lwronski** [comment]
It should be fixed by #1084.

But if  users use the latest stable scala version `3.1.2`,  the following error should be printed:
```
Starting compilation server
Compiling project (Scala 3.1.2, JVM)
Error: 18 is not a valid choice for -java-output-version
  scalac -help  gives more information
Error compiling project (Scala 3.1.2, JVM)
```

it seems, that scala compiler `3.1.2` doesn't support JVM `18`, so users should use the nightly version of scala compiler, it should work
```
$ /Users/lwronski/projects/scala-cli/out/cli/base-image/nativeImage.dest/scala-cli Hello.scala -S 3.nightly
Compiling project (Scala 3.2.0-RC1-bin-20220606-cec9aa3-NIGHTLY, JVM)
Compiled project (Scala 3.2.0-RC1-bin-20220606-cec9aa3-NIGHTLY, JVM)
```

**Comment by lwronski** [comment]
Fixed by #1084
